### PR TITLE
Make email.policy classes generic

### PR DESCRIPTION
**Pull request by srittau**
Cf. #10762

### CODE REVIEW

File: stdlib/email/__init__.pyi [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 # Definitions imported by multiple submodules in typeshed
 _ParamType: TypeAlias = str | tuple[str | None, str | None, str]  # noqa: Y047
 _ParamsType: TypeAlias = str | None | tuple[str, str | None, str]  # noqa: Y047
 
-def message_from_string(s: str, _class: Callable[[], Message] = ..., *, policy: Policy = ...) -> Message: ...
-def message_from_bytes(s: bytes | bytearray, _class: Callable[[], Message] = ..., *, policy: Policy = ...) -> Message: ...
-def message_from_file(fp: IO[str], _class: Callable[[], Message] = ..., *, policy: Policy = ...) -> Message: ...
-def message_from_binary_file(fp: IO[bytes], _class: Callable[[], Message] = ..., *, policy: Policy = ...) -> Message: ...
+# TODO: These functions should be generic over the message with appropriate
+# overloads.
```
Comment (AlexWaygood) [inline stdlib/email/__init__.pyi:12]: Would `Incomplete` be better than `Any`, then, in that case?

**Comment by Avasam** [comment]
@srittau Since you're currently touching a similar area in #11732 , just letting you know this now has conflicts.

**Comment by srittau** [comment]
I need to look at this in depth at some point.

**Comment by JelleZijlstra** [comment]
@srittau ping. There's a fairly big merge conflict too.

**Comment by srittau** [comment]
Redone as #12724.

**Pull request closed without merging**